### ISSUE TITLE
GSAM checkpoints don't appear to be valid .npz files

### ISSUE THREAD

**Issue by rwightman** [google-research/vision_transformer#195]
I can't load the .npz GSAM checkpoints in numpy. 

Also, if/when you get a chance to fix, would it be possible to name the files differently from the SAM npz files in the case where checkpoints are cached in a common folder? ie `ViT-B_16-GSAM.npz` would be preferred over both SAM and GSAM checkpoints being `ViT-B_16.npz`

@tingliu

**Comment by andsteing** [comment]
There was indeed an error with the `npz` conversion, thanks for flagging this!

As for model naming: note that we've consistently been using the same basename for different models, e.g.
- [`gs://vit_models/imagenet21k/ViT-B_16.npz`](https://console.cloud.google.com/storage/browser/vit_models/imagenet21k/ViT-B_16.npz)
- [`gs://vit_models/imagenet21k+imagenet2012/ViT-B_16.npz`](https://console.cloud.google.com/storage/browser/vit_models/imagenet21k+imagenet2012/ViT-B_16.npz)

These checkpoints can be downloaded from different directories (or specified via [`--config.pretrained_dir`](https://github.com/google-research/vision_transformer/blob/0dfc05ed4edf6aaf214974fbc08ab85750ee2b6c/vit_jax/configs/common.py#L27) and then the code expects the basename to be the same for every model with the same architecture:

https://github.com/google-research/vision_transformer/blob/0dfc05ed4edf6aaf214974fbc08ab85750ee2b6c/vit_jax/train.py#L115-L116

https://github.com/google-research/vision_transformer/blob/0dfc05ed4edf6aaf214974fbc08ab85750ee2b6c/vit_jax/configs/models.py#L288

So for that second point, I think it would be preferable to keep our original naming scheme and update the code in `timm` in a way to make sure that the models are cached correctly.

... but when I tried to run [`vit_jax.ipynb`] with the updated checkpoints, I found some new errors:

For `gsam/ViT_*` it seems that the models are missing the `CLS` token:
![image](https://user-images.githubusercontent.com/16050001/173565380-494035b6-a5cf-4e31-8f8c-e0ba86c8b3eb.png)

For `gsam/Mixer_*` it seems that the layers are named incorrectly:
![image](https://user-images.githubusercontent.com/16050001/173566527-46a5f25e-8bef-4729-99a0-96cf4051c1e8.png)

@tingliu can you look into this? probably best to verify the checkpoints with [`vit_jax.ipynb`] to verify they load correctly and reproduce expected accuracy after fine-tuning

[`vit_jax.ipynb`]: (https://colab.research.google.com/github/google-research/vision_transformer/blob/main/vit_jax.ipynb)

**Comment by rwightman** [comment]
@andsteing thanks, got the vit ones working w/ class token off and avg pool enabled.

**Comment by tingliu** [comment]
Thanks for reporting the issue and sorry for the delay. We have fixed the keys in the Mixer checkpoints, and now they should be compatible with the provided `vit_jax.ipynb`.
